WeatherXM/wxm-android
# FE-546 FE-542 Feature/bug fixes

## **Why?**
Fixed two minor bugs affecting the station's friendly name flow and the follow & historical charts flow.

### **How?**
1. Update the `CurrentViewModel`'s `device` when an updated one comes from the parent.
2. Update device's friendly name on `DeviceSettingsViewModel` when a success on editing or clearing the friendly name takes place.

### **Testing**
1. Make sure the bug as presented [here in the video](https://linear.app/weatherxm/issue/FE-542/research-historical-bugs-reported-by-george) is not reproducable anymore.
3. Make sure the bug as presented [here in the description](https://linear.app/weatherxm/issue/FE-546/cannot-rename-a-station-twice) is not reproducable anymore
